Engine/DataFeeds/CollectionSubscriptionDataSourceReader.cs

```diff
@@ -115,7 +115,10 @@ public IEnumerable<BaseData> Read(SubscriptionDataSource source)
                     catch (Exception err)
                     {
                         OnReaderError(raw, err);
-                        continue;
+                        if (!reader.RateLimit)
+                        {
+                            continue;
+                        }
```

Comment:
The current implementations of `RateLimit` only return `true` in live mode, so L130 would never be reached with `instances == null`. Currently not a bug, but just to be proactive, maybe better add a null check on `instances` before L130?

Reply:
Okay, I'll add the null check just in case 👍